core/src/test/java/org/apache/calcite/sql2rel/RelFieldTrimmerTest.java

```diff
@@ -672,5 +672,52 @@ public static Frameworks.ConfigBuilder config() {
     assertThat(trimmed, hasTree(expected));
   }
 
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3772">[CALCITE-3772]
+   * RelFieldTrimmer incorrectly trims fields when the query includes correlated-subquery</a>.
```

Comment:
Unit test are always welcome.
Can you please add a related link in JIRA [CALCITE-3772](https://issues.apache.org/jira/browse/CALCITE-3772), so reviewers can take a quick glance if they are interested in how this issue is resolved.

Reply:
I will check this.

Reply:
Fixed by CALCITE-5127 and add the link in Jira.

Reply:
Thanks for doing this.